src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/DropColumns.cs

```diff
@@ -74,24 +78,53 @@ public override bool CheckInvocation(TexlBinding binding, TexlNode[] args, DType
 
                 // Verify we have a string literal for the column name. Accd to spec, we don't support
                 // arbitrary expressions that evaluate to string values, because these values contribute to
-                // type analysis, so they need to be known upfront (before DropColumns executes).
-                StrLitNode nameNode;
-                if (nameArgType.Kind != DKind.String || (nameNode = nameArg.AsStrLit()) == null)
+                // type analysis, so they need to be known upfront (before DropColumns executes).                            
+                if (supportIndentifiers)
                 {
-                    fArgsValid = false;
-                    errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrExpectedStringLiteralArg_Name, nameArg.ToString());
-                    continue;
+                    FirstNameNode identifierNode = nameArg.AsFirstName();
+
+                    if (identifierNode == null)
+                    {
+                        fArgsValid = false;
+
+                        // Argument '{0}' is invalid, expected an identifier.
+                        errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrExpectedIdentifierArg_Name, nameArg.ToString());
+                        continue;
+                    }
+
+                    value = identifierNode.Ident.Name;
+                }
+                else
+                {
+                    StrLitNode strLitNode = nameArg.AsStrLit();
+
+                    if (nameArgType.Kind != DKind.String && strLitNode == null)
+                    {
+                        fArgsValid = false;
+
+                        // Argument '{0}' is invalid, expected a text literal.
+                        errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrExpectedStringLiteralArg_Name, nameArg.ToString());
+                        continue;
+                    }
+
+                    value = strLitNode.Value;
                 }
 
                 // Verify that the name is valid.
-                if (!DName.IsValidDName(nameNode.Value))
+                if (!DName.IsValidDName(value))
                 {
                     fArgsValid = false;
-                    errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrArgNotAValidIdentifier_Name, nameNode.Value);
+
+                    // Argument '{0}' is not a valid identifier.
+                    errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrArgNotAValidIdentifier_Name, value);
                     continue;
                 }
 
-                var columnName = new DName(nameNode.Value);
+                DName? displayName;
+
+                var columnName = nameArg is FirstNameNode firstNameNode && ((displayName = binding.GetDisplayName(firstNameNode)) != null)
+                    ? displayName.Value
+                    : new DName(value);
```

Comment:
this seems incorrect: display names aren't in the type, by definition. Those are only logical names.

Reply:
This is how I set the displayname in the type.

            var rt = RecordType.Empty()
                .Add(new NamedFormulaType("logicalA", FormulaType.Number, displayName: "displayName"))
                .Add(new NamedFormulaType("logicalB", FormulaType.Number));

Help me fixing the code if it's incorrect, I don't know what to do.